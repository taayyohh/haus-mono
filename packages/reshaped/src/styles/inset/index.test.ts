import getInsetStyles from "./index";

describe("Styles/Inset", () => {
	test("handles positive value", () => {
		expect(getInsetStyles(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(getInsetStyles(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(getInsetStyles()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(getInsetStyles({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});

	test("handles value width defined side", () => {
		expect(getInsetStyles(4, "start")).toMatchSnapshot();
	});
});
