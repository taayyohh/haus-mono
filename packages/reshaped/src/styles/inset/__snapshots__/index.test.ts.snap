// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/Inset > handles 0 value 1`] = `
{
  "classNames": [
    "_--inset_a9394a",
  ],
  "variables": {
    "--rs-inset-s": 0,
  },
}
`;

exports[`Styles/Inset > handles positive value 1`] = `
{
  "classNames": [
    "_--inset_a9394a",
  ],
  "variables": {
    "--rs-inset-s": 4,
  },
}
`;

exports[`Styles/Inset > handles responsive value 1`] = `
{
  "classNames": [
    "_--inset_a9394a",
    "_--inset--m_a9394a",
    "_--inset--l_a9394a",
  ],
  "variables": {
    "--rs-inset-l": 2,
    "--rs-inset-m": 0,
    "--rs-inset-s": 4,
  },
}
`;

exports[`Styles/Inset > handles undefined value 1`] = `null`;

exports[`Styles/Inset > handles value width defined side 1`] = `
{
  "classNames": [
    "_--inset-start_a9394a",
  ],
  "variables": {
    "--rs-inset-start-s": 4,
  },
}
`;
